### PR TITLE
Ciac 11383

### PR DESCRIPTION
**Pull request by mayyagoldman**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-11383?filter=-1)

## Description
A new argument, includeinformational (Boolean), is being added to the getIncidents API. The possible inputs are true or false, with the default set to false.

In XSIAM, this argument will be included as an input. When set to true, it allows the retrieval of “Info” severity alerts alongside other severities.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/ReleaseNotes/1_15_68.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+#### Scripts
+
+##### SearchIncidentsV2
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/ReleaseNotes/1_15_68.md:4]: Please edit this release note so it will be relevant to XSIAM only

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.yml [hunk 2]
```diff
@@ -64,6 +64,14 @@ args:
   predefined:
   - "false"
   - "true"
+- description: When the value is set to 'True', informational severity alerts will return as part of the results. The ‘fromdate’ and ‘todate’ arguments must be provided to use this argument. The maximum value currently supported for the 'fromdate' argument to retrieve informational incidents is 5 hours. If a value greater than this is provided, it will be adjusted to 5 hours ago. To retrieve only informational incidents, use the `query` argument and include this limitation within the query. Default is false.
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.yml:67]: Why don't you use default value = false?

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/README.md [hunk 3]
```diff
@@ -60,31 +60,32 @@ Spring Core and Cloud Function SpEL RCEs
...
+| searchresultslabel | If provided, the value of this argument will be set under the searchResultsLabel context key for each incident found.                                                                                                                              |
 | summarizedversion  | If enabled runs a summarized version of this script. Disables auto-extract, sets fromDate to 30 days, and minimizes the context output. You can add sepcific fields to context using the add_fields_to_summarize_context argument. Default is false. |
-| limit              | The maximum number of incidents to be returned. Default is 100.                                                                                                                                                                                      |
+| includeinformational | When the value is set to  'True', informational severity alerts will return as part of the results. Default is false.                                                                                                                              |
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/README.md:87]: This description was edited changed accordingly.

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py [hunk 4]
```diff
@@ -7,6 +7,9 @@
 DEFAULT_PAGE_SIZE = 100
 STARTING_PAGE_NUMBER = 1
 
+INCLUDEINFORMATIONAL_SUPPORT_VERSION = '8.8'
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py:10]: Didnt we say we remove the version check?

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py [hunk 5]
```diff
@@ -142,6 +145,18 @@ def search_incidents(args: Dict):   # pragma: no cover
         if args.get('trimevents') == '0':
             args.pop('trimevents')
 
+    if args.get('includeinformational'):
+        if not ENABLE_INCLUDEINFORMATIONAL_ARG:
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py:149]: Why did you keep this in the code?

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py [hunk 6]
```diff
@@ -142,6 +145,18 @@ def search_incidents(args: Dict):   # pragma: no cover
...
+            raise ValueError('The includeinformational argument supported only in XSIAM from version x.x.x.')
+        if not (args.get('fromdate') and args.get('todate')):
+            raise ValueError('The includeinformational argument requires fromdate and todate arguments.')
+        if (datetime.utcnow() - fromdate).total_seconds() > 5 * 60 * 60:
+            args['fromdate'] = arg_to_datetime('5 hours ago').isoformat()
+            demisto.log(f'fromdate: {fromdate} is more than 5 hours from now: {datetime.utcnow()}. '
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py:155]: We use demisto.info or demisto.debug (This will not be presented to the user unless he will look at the logs. Please change this to a readable output.

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/README.md [hunk 7]
```diff
@@ -60,31 +60,32 @@ Spring Core and Cloud Function SpEL RCEs
...
-| size               | Number of incidents per page \(per fetch\) (deprecated)                                                                                                                                                                                              |
-| sort               | Sort in format of field.asc,field.desc,...                                                                                                                                                                                                           |
-| searchresultslabel | If provided, the value of this argument will be set under the searchResultsLabel context key for each incident found.                                                                                                                                |
+| **Argument Name**  | **Description**                                                                                                                                                                                                                                    |
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/README.md:63]: Why was all the table changed please revert all changes that are not related to your modification.

**Review by omerKarkKatz — CHANGES_REQUESTED** [review]
Hey @mayyagoldman nice job! Please review the comments,
1. Add UT 
2. Assignee some one from the technical writer team.

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py (hunk 6 above)
Comment (mayyagoldman) [reply Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py:163]: when I use demisto.info I dont see the log in the war room UI, are you sure that this needs to be changed? 
demisto.log seems to produce the desired result

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/README.md [hunk 8]
```diff
@@ -238,4 +239,3 @@ Spring Core and Cloud Function SpEL RCEs
 |id|name|severity|status|owner|created|closed|
 |---|---|---|---|---|---|---|
 | 978 | Incident to search | 0 | 0 | admin | 2020-09-29T17:29:44.162034+03:00 | 0001-01-01T00:00:00Z |
-
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/README.md:241]: Please revert

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py [hunk 9]
```diff
@@ -155,16 +156,30 @@ def search_incidents(args: Dict):   # pragma: no cover
...
 
+    if includeinformational := argToBoolean(args.get('includeinformational', False)):
+        if not (args.get('fromdate') and args.get('todate')):
+            raise ValueError('The includeinformational argument requires fromdate and todate arguments.')
+        if (datetime.utcnow() - fromdate).total_seconds() > 5 * 60 * 60:  # type: ignore
+            args['fromdate'] = arg_to_datetime('5 hours ago').isoformat()  # type: ignore
+            hr_prefix = (f'fromdate: {fromdate} is more than 5 hours from now: {datetime.utcnow()}. Currently, we support '
+                         f'querying informational incidents for up to the last 5 hours. The fromdate has been'
+                         f' adjusted to {args.get("fromdate")}')
+        args['includeinformational'] = includeinformational
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py:167]: Why do we need this line?

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py [hunk 10]
```diff
@@ -155,16 +156,30 @@ def search_incidents(args: Dict):   # pragma: no cover
...
+    else:
+        args.pop('includeinformational', None)
+
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py:170]: Our standards to add new lines after if/else cluase.
```suggestion
    if includeinformational := argToBoolean(args.get('includeinformational', False)):
        if not (args.get('fromdate') and args.get('todate')):
            raise ValueError('The includeinformational argument requires fromdate and todate arguments.')

        if (datetime.utcnow() - fromdate).total_seconds() > 5 * 60 * 60:  # type: ignore
            args['fromdate'] = arg_to_datetime('5 hours ago').isoformat()  # type: ignore
            hr_prefix = (f'fromdate: {fromdate} is more than 5 hours from now: {datetime.utcnow()}. Currently, we support '
                         f'querying informational incidents for up to the last 5 hours. The fromdate has been'
                         f' adjusted to {args.get("fromdate")}')
        args['includeinformational'] = includeinformational

    else:
        args.pop('includeinformational', None)

```

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2_test.py [hunk 11]
```diff
@@ -1,5 +1,6 @@
 from SearchIncidentsV2 import *
 import pytest
+import datetime as dt
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2_test.py:4]: Please add given when then to all your tests.

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py [hunk 12]
```diff
@@ -155,16 +156,30 @@ def search_incidents(args: Dict):   # pragma: no cover
         if args.get('trimevents') == '0':
             args.pop('trimevents')
 
+    if includeinformational := argToBoolean(args.get('includeinformational', False)):
+        if not (args.get('fromdate') and args.get('todate')):
+            raise ValueError('The includeinformational argument requires fromdate and todate arguments.')
+        if (datetime.utcnow() - fromdate).total_seconds() > 5 * 60 * 60:  # type: ignore
+            args['fromdate'] = arg_to_datetime('5 hours ago').isoformat()  # type: ignore
+            hr_prefix = (f'fromdate: {fromdate} is more than 5 hours from now: {datetime.utcnow()}. Currently, we support '
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py:164]:
```suggestion
            hr_prefix = (f'fromdate: {fromdate} is more than 5 hours ago. We support '
```

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py [hunk 13]
```diff
@@ -155,16 +156,30 @@ def search_incidents(args: Dict):   # pragma: no cover
...
+                         f'querying informational incidents for up to the last 5 hours. The fromdate has been'
+                         f' adjusted to {args.get("fromdate")}')
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py:166]: If ther user will get the time in UTC it can confuse him i think we better use the relative time "5 hours afo"

File: Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py (hunk 9 above)
Comment (mayyagoldman) [reply Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py:167]: 167? 
update it to boolean rather than string
